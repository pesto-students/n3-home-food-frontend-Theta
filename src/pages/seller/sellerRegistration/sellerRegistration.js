import { Button, notification, Col, Form, Input, Row } from "antd";
import "antd/dist/antd.css";
import { React } from "react";
import { connect, useDispatch } from "react-redux";

import "./sellerRegistration.css";
import CustomerLogin from "../../landingScreen/customerLogin";
import { setIsCustomerLoginDrawerOpen } from "../../../store/actions";
import axios from "../../../utils/axios";

function SellerRegistration() {
  const Dispatch = useDispatch();

  const toggleDrawer = () => {
    Dispatch(setIsCustomerLoginDrawerOpen(true));
  };

  const registerSeller = async (form) => {
    let response = await axios
      .post("/sellers/register", form)
      .then((response) => {
        notification.success({
          message: "Great",
          description: "Successfully Register login with your mobile number",
          placement: "topLeft",
        });
        setTimeout(() => {
          toggleDrawer();
        }, [2000]);
      })
      .catch((error) => {
        return error.response;
      });
    return response;
  };

  return (
    <>
      <Row style={{ height: "100vh" }}>
        <Col md={16} className="left-container">
          <div className="left-container-content">
            <h1>Partner with us</h1>
            <p>
              What is Lorem Ipsum Lorem Ipsum is simply dummy text of the
              printing and typesetting industry Lorem Ipsum has been the
              industry's standard dummy text ever since the 1500s when an
              unknown printer took a galley of type and scrambled it to make a
              type specimen book it has?
            </p>
          </div>
        </Col>
        <Col md={8}>
          <div className="right-container">
            <h4 align="center">Register</h4>
            <Form layout="vertical" hideRequiredMark onFinish={registerSeller}>
              <Row gutter={16}>
                <Col span={24}>
                  <Form.Item
                    name="name"
                    label="Name"
                    rules={[
                      { required: true, message: "Please Enter User Name" },
                    ]}
                  >
                    <Input placeholder="Please Enter User Name" />
                  </Form.Item>
                </Col>
                <Col span={24}>
                  <Form.Item
                    name="display_name"
                    label="Display Name"
                    rules={[
                      { required: true, message: "Please Enter Display Name" },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder="Please Enter Display Name"
                    />
                  </Form.Item>
                </Col>

                <Col span={24}>
                  <Form.Item
                    name="phone"
                    label="Mobile Number"
                    rules={[
                      { required: true, message: "Please Enter Mobile Number" },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder="Please enter Mobile Number"
                    />
                  </Form.Item>
                </Col>
              </Row>
              <Row gutter={16}>
                <Col span={12}>
                  <Form.Item
                    name="idProof"
                    label="Aadhaar Card"
                    rules={[
                      { required: true, message: "Please Enter Aadhaar Card" },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder="Please Enter Aadhaar Card"
                    />
                  </Form.Item>
                </Col>

                <Col span={12}>
                  <Form.Item
                    name="email"
                    label="Email (optional)"
                    rules={[{ required: false, message: "Please Enter Email" }]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder="Please Enter Email"
                    />
                  </Form.Item>
                </Col>
              </Row>

              <Row gutter={16}>
                <Col span={24}>
                  <Form.Item
                    name="adress"
                    label="Address"
                    rules={[
                      {
                        required: true,
                        message: "Please Enter Address",
                      },
                    ]}
                  >
                    <Input.TextArea
                      rows={4}
                      placeholder="Please Enter Address"
                    />
                  </Form.Item>
                </Col>
              </Row>

              <CustomerLogin type="Seller" />
              <Row gutter={[13, 13]} justify="end">
                <Button block type="primary" htmlType="submit">
                  Submit
                </Button>
                <Button onClick={toggleDrawer} block type="primary">
                  Login
                </Button>
              </Row>
            </Form>
          </div>
        </Col>
      </Row>
    </>
  );
}

const mapStateToProps = (state) => {
  return {
    title: state,
  };
};
export default connect(mapStateToProps)(SellerRegistration);
