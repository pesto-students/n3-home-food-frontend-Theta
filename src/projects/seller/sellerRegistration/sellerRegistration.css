.left-container{
 height: 100vh;
 background:#000000d9;
 display: flex;
 align-items: center;
}
.left-container-content{
    padding: 40px;
    width: 80%;
    height: auto;
    color: #fff;
}
.left-container-content h1{
    border-bottom: 2px solid;
    padding-bottom: 26px;
    color: #fff;
    font-family: system-ui;
    width: 70%;
}
.right-container{
    padding: 20px 50px;
}
