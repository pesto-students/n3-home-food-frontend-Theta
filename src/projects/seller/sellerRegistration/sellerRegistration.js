import { Button, notification, Col, Form, Input, Row } from "antd";

import { React } from "react";
import { connect, useDispatch } from "react-redux";
import { useTranslation } from "react-i18next";

import "./sellerRegistration.css";
import CustomerLogin from "landingScreen/customerLogin";
import { setIsCustomerLoginDrawerOpen } from "store/actions";
import { registerSeller } from "../utils/api";
import { catchError } from "utils/helpers";

function SellerRegistration() {
  const { t } = useTranslation();

  const Dispatch = useDispatch();

  const toggleDrawer = () => {
    Dispatch(setIsCustomerLoginDrawerOpen(true));
  };

  const newAccount = async (form) => {
    try {
      const response = await registerSeller(form);
      if (response.status === 400) {
        notification.error({
          message: "Error",
          description: response.data,
          placement: "topLeft",
        });
        return;
      }
      if (response.status === 200) {
        notification.success({
          message: "Great",
          description: "Successfully Register login with your mobile number",
          placement: "topLeft",
        });
        setTimeout(() => {
          toggleDrawer();
        }, [2000]);
      }
    } catch (error) {
      catchError(error);
    }
  };

  return (
    <>
      <Row style={{ height: "100vh" }}>
        <Col md={16} className="left-container">
          <div className="left-container-content">
            <h1> {t("seller.registerForm.partnerText")}</h1>
            <p>{t("seller.registerForm.description")}</p>
          </div>
        </Col>
        <Col md={8}>
          <div className="right-container">
            <h4 align="center">{t("seller.registerForm.register")}</h4>
            <Form layout="vertical" hideRequiredMark onFinish={newAccount}>
              <Row gutter={16}>
                <Col span={24}>
                  <Form.Item
                    name="name"
                    label={t("seller.registerForm.name")}
                    rules={[
                      {
                        required: true,
                        message: t(
                          "seller.registerForm.nameValidationAndPlaceholder"
                        ),
                      },
                    ]}
                  >
                    <Input
                      placeholder={t(
                        "seller.registerForm.nameValidationAndPlaceholder"
                      )}
                    />
                  </Form.Item>
                </Col>
                <Col span={24}>
                  <Form.Item
                    name="display_name"
                    label={t("seller.registerForm.displayName")}
                    rules={[
                      {
                        required: true,
                        message: t(
                          "seller.registerForm.displayNameValidationAndPlaceholder"
                        ),
                      },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder={t(
                        "seller.registerForm.displayNameValidationAndPlaceholder"
                      )}
                    />
                  </Form.Item>
                </Col>

                <Col span={24}>
                  <Form.Item
                    name="phone"
                    label={t("seller.registerForm.mobileNumber")}
                    rules={[
                      {
                        required: true,
                        message: t(
                          "seller.registerForm.mobileNumberValidationAndPlaceholder"
                        ),
                      },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder={t(
                        "seller.registerForm.mobileNumberValidationAndPlaceholder"
                      )}
                    />
                  </Form.Item>
                </Col>
              </Row>
              <Row gutter={16}>
                <Col span={12}>
                  <Form.Item
                    name="idProof"
                    label={t("seller.registerForm.idProof")}
                    rules={[
                      {
                        required: true,
                        message: t(
                          "seller.registerForm.idProofValidationAndPlaceholder"
                        ),
                      },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder={t(
                        "seller.registerForm.idProofValidationAndPlaceholder"
                      )}
                    />
                  </Form.Item>
                </Col>

                <Col span={12}>
                  <Form.Item
                    name="email"
                    label={t("seller.registerForm.email")}
                    rules={[
                      {
                        required: false,
                        message: t(
                          "seller.registerForm.emailValidationAndPlaceholder"
                        ),
                      },
                    ]}
                  >
                    <Input
                      style={{ width: "100%" }}
                      placeholder={t(
                        "seller.registerForm.emailValidationAndPlaceholder"
                      )}
                    />
                  </Form.Item>
                </Col>
              </Row>

              <Row gutter={16}>
                <Col span={24}>
                  <Form.Item
                    name="adress"
                    label={t("seller.registerForm.address")}
                    rules={[
                      {
                        required: true,
                        message: t(
                          "seller.registerForm.addressValidationAndPlaceholder"
                        ),
                      },
                    ]}
                  >
                    <Input.TextArea
                      rows={4}
                      placeholder={t(
                        "seller.registerForm.addressValidationAndPlaceholder"
                      )}
                    />
                  </Form.Item>
                </Col>
              </Row>

              <CustomerLogin type="Seller" />
              <Row gutter={[13, 13]}>
                <Button block type="primary" htmlType="submit">
                  {t("seller.registerForm.submit")}
                </Button>
              </Row>
              <Row gutter={24} className="mt-3">
                <Col span={6}></Col>
                <Col span={12}>
                  {/* <Link onClick={toggleDrawer} block type="primary">
                    <span> {t("seller.registerForm.loginText")}</span>
                  </Link> */}
                </Col>
              </Row>
            </Form>
          </div>
        </Col>
      </Row>
    </>
  );
}

const mapStateToProps = (state) => {
  return {
    title: state,
  };
};
export default connect(mapStateToProps)(SellerRegistration);
