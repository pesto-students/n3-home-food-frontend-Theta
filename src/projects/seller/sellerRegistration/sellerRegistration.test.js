import * as React from "react";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import sellerRegistration from "./sellerRegistration";

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Seller Registration Dashboard component", () => {
  let wrapper;
  test("render without error", () => {
    wrapper = shallow(<sellerRegistration />);
    expect(wrapper.length).toBe(1);
  });

  test('should render correctly in "debug" mode', () => {
    const component = shallow(<sellerRegistration />);

    expect(component).toMatchSnapshot();
  });
});