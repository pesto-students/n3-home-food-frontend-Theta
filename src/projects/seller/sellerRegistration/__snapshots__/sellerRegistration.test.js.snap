// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seller Registration Dashboard component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
